git/objects/util.py

```diff
@@ -568,11 +568,11 @@ def addToStack(
                 yield rval
 
             # Only continue to next level if this is appropriate!
-            nd = d + 1
-            if depth > -1 and nd > depth:
+            nb = d + 1
```

Comment:
Why does it do that?

Reply:
this is probably left-over from previously used tool